independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartSpec.java

```diff
@@ -0,0 +1,172 @@
+package io.quarkus.devtools.codestarts;
+
+import static java.util.Objects.requireNonNull;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public final class CodestartSpec {
+
+    enum Type {
+        PROJECT(true),
+        LANGUAGE(true),
+        BUILDTOOL(true),
+        CONFIG(true),
+        EXAMPLE(false),
+        TOOLING(false);
+
+        private final boolean base;
+
+        Type(boolean base) {
+            this.base = base;
+        }
+
+        public boolean isBase() {
+            return base;
+        }
+    }
+
+    private final String name;
+    private final boolean isPreselected;
+    private final String ref;
+    private final Type type;
+    private final boolean isFallback;
+    private final List<String> missingLanguages;
+    private final Map<String, String> outputStrategy;
+    private final Map<String, LanguageSpec> languagesSpec;
+
+    @JsonCreator
+    public CodestartSpec(@JsonProperty(value = "name", required = true) String name,
+            @JsonProperty(value = "ref") String ref,
+            @JsonProperty(value = "type") Type type,
+            @JsonProperty("fallback") boolean isFallback,
+            @JsonProperty("preselected") boolean isPreselected,
+            @JsonProperty(value = "missing-languages") List<String> missingLanguages,
+            @JsonProperty("output-strategy") Map<String, String> outputStrategy,
+            @JsonProperty("language") Map<String, LanguageSpec> languagesSpec) {
+        this.name = requireNonNull(name, "name is required");
```

Comment:
when that fails - will users get some context of which codestart spec that is failing ?

Reply:
I will make sure of that